src/hotspot/share/nmt/memReporter.hpp

```diff
@@ -149,11 +150,11 @@ class MemSummaryReporter : public MemReporterBase {
 class MemDetailReporter : public MemSummaryReporter {
  private:
   MemBaseline&   _baseline;
-
+  NativeCallStackPrinter _stackprinter;
  public:
   MemDetailReporter(MemBaseline& baseline, outputStream* output, size_t scale = default_scale) :
     MemSummaryReporter(baseline, output, scale),
-     _baseline(baseline) { }
+     _baseline(baseline), _stackprinter(output) { }
```

Comment:
`NativeCallStackPrinter` is 293* 1024 == 293KB

Isn't there a way to create/destroy it on the fly? NMT is already memory hungry, we just keep increasing the price of admission ...

Reply:
MemDetailReporter is only created for a detail report and does not live beyond reporting. The printer cache is needed for the duration of the report. Simplest way to do this is an inline member in the reporter class.